### PR TITLE
soc: nxp: rw6xx: enable DMIC support

### PR DESCRIPTION
**Pull request by danieldegrasse**
Enable support for the DMIC PDM microphone IP on the RW6xx and RD RW612 BGA board. Support was verified with a PDM generator in mono and stereo channel mode.

### CODE REVIEW

**Review by decsny — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

File: boards/nxp/rd_rw612_bga/rd_rw612_bga.dtsi [hunk 1]
```diff
@@ -125,6 +126,47 @@ arduino_i2c: &flexcomm2 {
...
+&dmic0 {
+	pinctrl-0 = <&pinmux_dmic0>;
+	pinctrl-names = "default";
+	use2fs;
+	status = "okay";
+};
+
+/* Configure pdm channels 0-3 with gain, and cutoff settings
+ * appropriate for the attached MEMS microphones.
+ * Note that PDCM0/PDMC1 and PDMC2/PDMC3 are each connected to the same
```
Comment (kartben) [inline boards/nxp/rd_rw612_bga/rd_rw612_bga.dtsi:138]:
```suggestion
 * Note that PDMC0/PDMC1 and PDMC2/PDMC3 are each connected to the same
```

**Review by kartben — DISMISSED** [review]

File: boards/nxp/rd_rw612_bga/rd_rw612_bga.dtsi (hunk 1 above)
Comment (decsny) [reply boards/nxp/rd_rw612_bga/rd_rw612_bga.dtsi:138]: you have a hawkeye for typos

**Review by dleach02 — APPROVED** [review]

**Comment by kartben** [comment]
@danieldegrasse @dleach02 @decsny could you guys please make sure this board is covered in the MAINTAINERS file. I am the assignee because it touches a sample, but it doesn't sound right (will re-assign to @dleach02)

**Comment by decsny** [comment]
> @danieldegrasse @dleach02 @decsny could you guys please make sure this board is covered in the MAINTAINERS file. I am the assignee because it touches a sample, but it doesn't sound right (will re-assign to @dleach02)

@kartben  the board is covered, I guess it's because of the sample overlay, you are maintainer of samples, but the bot seems to be getting this totally wrong, and I noticed a lot of our PRs that change seemingly 99% NXP code get assigned to someone else because of some tiny overlay, @nashif can you help figure this out

**Comment by kartben** [comment]
> @kartben the board is covered

Ah yes, sorry, my bad. I was checking on a branch that was slightly too old and missed it.

**Comment by dleach02** [comment]
@kartben I'm reluctant to change the assignee to me in many cases but will in this case. If you find additional PRs going your way and don't feel it is correct then please feel free to reassign to me.

**Review by decsny — APPROVED** [review]

**Pull request merged**